### ISSUE TITLE
Path encoding not preserved in JAX-RS

### ISSUE THREAD

**Issue by m0mus** [oracle/helidon#150]
JAX-RS matches @Path annotations against URL-encoded path. But the encoding is not preserved.
Sample resource:
```
@Path("/res")
@ApplicationScoped
public class TestResource {

    @Path("/{id:[^/_]*}/hello")
    @GET
    public String test1(@PathParam("id") String param) {
        return param;
    }

    @Path("/{id}")
    @GET
    public String test2(@PathParam("id") String param) {
        return param;
    }

}
```

Try the following requests:

|Request|Actual Response|Expected Response|
|--------|-----------------|--------------------|
|localhost:7001/res/abc%3Bd/hello|abc|abc;d|
|localhost:7001/res/abc%2Fd| 404 status code | abc/d|

This is caused by reading (and passing) the decoded path from the URI instead of the raw path (in RequestRouting class):
```
 // Process path
String p = bareRequest.getUri().normalize().getPath();
```
This is then re-encoded before passing it to Jersey.

Reported by Martin Sladecek

**Comment by romain-grecourt** [comment]
@tomas-langer @barchetta I'm targeting this one for 1.0

**Comment by romain-grecourt** [comment]
Fixed by #317
